proto/zitadel/user/v3alpha/authenticator.proto

```diff
@@ -0,0 +1,468 @@
+  // A list of the user's linked identity providers (IDPs).
+  repeated IdentityProvider identity_providers = 8;
+}
+
+message Username {
+  // unique identifier of the username.
+  string username_id = 1;
+  // The user's unique username. It is used for identification during authentication.
+  string username = 2 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"gigi-giraffe\"";
+    }
+  ];
+  // By default usernames must be unique across all organizations in an instance.
+  // This option allow to restrict the uniqueness to the user's own organization.
+  // As a result, this username can only be used if the authentication is limited
+  // to the corresponding organization.
+  //
+  // This can be useful if you provide multiple usernames for a single user, where one
+  // if specific to your organization, e.g.:
+  // - gigi-giraffe@zitadel.com (unique across organizations)
+  // - gigi-giraffe (unique only inside the ZITADEL organization)
+  bool is_organization_specific = 3;
+}
+
+message SetUsername {
+  // Set the user's username. This will be used for identification during authentication.
+  string username = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1,
+      max_length: 200,
+      example: "\"gigi-giraffe\"";
+    }
+  ];
+  // By default username must be unique across all organizations in an instance.
+  // This option allow to restrict the uniqueness to the user's own organization.
+  // As a result, this username can only be used if the authentication is limited
+  // to the corresponding organization.
+  //
+  // This can be useful if you provide multiple usernames for a single user, where one
+  // if specific to your organization, e.g.:
+  // - gigi-giraffe@zitadel.com (unique across organizations)
+  // - gigi-giraffe (unique only inside the ZITADEL organization)
+  bool is_organization_specific = 2;
+}
+
+message Password {
+  // States the time the password was last changed.
+  //(--TODO:do we want to include that in the list or is there a separate call?--)
+  google.protobuf.Timestamp last_changed = 1 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"2019-04-01T08:45:00.000000Z\"";
+    }
+  ];
+}
+
+message WebAuthN {
+  // unique identifier of the WebAuthN authenticator.
+  string web_auth_n_id = 1 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"69629023906488334\""
+    }
+  ];
+  // Name of the WebAuthN authenticator. This is used for easier identification.
+  string name = 2 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"fido key\""
+    }
+  ];
+  // State whether the WebAuthN registration has been completed.
+  //(--TODO:only verified or return verification state?--)
+  bool is_verified = 3;
```

Comment:
general question for authenticator, which need to be verified:
do we only return verified or always the verification state?

Reply:
Do we have an authenticator with several states?

Reply:
nope. only some which have a two step registration process (webauthn, (t)otp, ...)
typically you would do the two calls of the registration just after each other and therefore know the authenticator ID required for the second call. If you however interrupt the registration, you would need to be able to retrieve it again. in the other hand, you could simply restart the process again as well

Reply:
Hm yeah I think a boolean would suffice, as internally we maybe want to discern the states but on the API the only information we need is that it would be usable.